src/components/AnchoredTextLabels.cpp

```diff
@@ -0,0 +1,61 @@
+#include "AnchoredTextLabels.h"
+#include <Components/BsCCamera.h>
+#include <GUI/BsCGUIWidget.h>
+#include <GUI/BsGUILabel.h>
+#include <GUI/BsGUIPanel.h>
+#include <Math/BsMath.h>
+#include <Scene/BsSceneObject.h>
+#include <excepction/Throw.hpp>
+
+namespace REGoth
+{
+  AnchoredTextLabels::AnchoredTextLabels(const bs::HSceneObject& parent)
+      : Component(parent)
+  {
+    // Set a name for the component, so we can find it later if needed
+    setName("AnchoredTextLabels");
+
+    // A GUI widget component is required for drawing the labels
+    mGui = parent->getComponent<bs::CGUIWidget>();
```

Comment:
Why are you taking the parents component? Couldn't we just add the  `AnchoredTextLabels` component to the same scene object?

Reply:
Or maybe even allow passing a handle to CGUI via the constructor. Then we could have multiple instances of the component managed by whatever needs it... Or should we just search up the parent chain?

Reply:
It is supposed to be added to a scene object which has a `CGUIWidget` component already. I mean you could definitely loosen the constraint and pass the required component in the constructor. This is the way I would favor, as it still allows the add the component to a "GUI scene object" or just any other scene object, if you e. g. have a scene object which manages all debug related drawing or something like that.

Reply:
> It is supposed to be added to a scene object which has a CGUIWidget component already. 

Then you don't want `parent`, but rather `SO()`, which is the same thing but it's the common notation for accessing a components scene object. There is also `SO()->parent()` or something which threw me off. I thought you were searching the scene object which is the parent to the scene object the component is attached to.

Reply:
`parent` is the scene object which owns the component. I see why you were confused though. I will add the `CGUIWidget` component as argument to the constructor. This should make things more clear.

Reply:
Done in 1b6da35.